src/d3a/models/state.py

```diff
@@ -249,6 +298,19 @@ def total_energy_demanded_Wh(self):
     def get_desired_energy(self, time_slot):
         return self._desired_energy_Wh[time_slot]
 
+    def _calculate_unsettled_energy_kWh(self, measured_energy_kWh, time_slot):
+        """
+        Returns negative values for underconsumption (offer will be placed on the settlement
+        market) and positive values for overconsumption (bid will be placed on the settlement
+        market)
+        :param measured_energy_kWh: Measured energy that the PV produced
+        :param time_slot: time slot of the measured energy
+        :return: Deviation between forecasted and measured energy
+        """
+        traded_energy_kWh = (self.get_desired_energy_Wh(time_slot) -
+                             self.get_energy_requirement_Wh(time_slot)) * 1000.0
```

Comment:
```suggestion
                             self.get_energy_requirement_Wh(time_slot)) / 1000.0
```
I think division is required to convert smaller unit to a bigger one.

Reply:
Nice catch, you are correct! Also adding unit tests to cover this case, as far as I see the state classes are all lacking. Thanks!